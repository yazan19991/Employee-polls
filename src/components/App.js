import { connect } from 'react-redux';
import { Routes, Route } from 'react-router-dom';
import { useEffect } from 'react';
import { handleReceiveUsers } from '../actions/users';
import { handleReceivePolls } from '../actions/polls';
import { handleReceiveAuthedUser } from '../actions/authedUser';
import LoginPage from './LoginPage';
import HomePage from './HomePage';
import PollPage from './PollPage';
import Leaderboard from './Leaderboard';
import CreatePage from './CreatePage';
import Nav from './Nav';

function App(props) {
  useEffect(() => {
    props.dispatch(handleReceiveUsers());
    // eslint-disable-next-line
  }, []);

  useEffect(() => {
    props.dispatch(handleReceivePolls());
    // eslint-disable-next-line
  }, []);

  useEffect(() => {
    props.dispatch(handleReceiveAuthedUser());
    // eslint-disable-next-line
  }, []);

  return (
    <div>
      <Nav />
      <Routes>
        <Route path="/" exact element={<LoginPage />} />
        <Route path="/login" exact element={<LoginPage />} />
        <Route path="/home" exact element={<HomePage />} />
        <Route path="/question/:id" exact element={<PollPage />} />
        <Route path="/leaderboard" exact element={<Leaderboard />} />
        <Route path="/add" exact element={<CreatePage />} />
      </Routes>
    </div>
  );
}

export default connect()(App);
