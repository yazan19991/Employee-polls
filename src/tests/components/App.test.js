import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../../components/App';
import { createStore } from 'redux';
import reducer from '../../reducers';
import middleware from '../../middleware';
import { MemoryRouter } from 'react-router-dom';

const store = createStore(reducer, middleware);

describe('Snapshot testing App component', () => {
  it('Will match the snapshot', () => {
    const { component } = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
