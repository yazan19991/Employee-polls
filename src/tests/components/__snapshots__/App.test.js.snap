// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`Snapshot testing App component Will match the snapshot 1`] = `undefined`;
